ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/MainnetRequestsValidator.java

```
@@ -41,20 +41,24 @@ public class MainnetRequestsValidator implements RequestsValidator {
    */
   @Override
   public boolean validate(final Optional<List<Request>> maybeRequests) {
-    if (maybeRequests.isEmpty()) {
-      LOG.warn("Must contain requests (even if empty list)");
+    if (containsRequestWithEmptyData(maybeRequests.get())) {
```

should we discard everything if at least one has empty data? From [eip-7685](https://eips.ethereum.org/EIPS/eip-7685) it is said to discard the ones that are empty but still use the other ones:
> In order to compute the commitment, an intermediate hash list is first built by hashing all non-empty requests elements of the block requests list. Items with empty request_data are excluded, i.e. the intermediate list skips requests items which contain only the request_type (1 byte) and nothing else.

---

The spec (https://github.com/ethereum/execution-apis/blob/main/src/engine/prague.md) says "Elements with empty request_data MUST be excluded from the list. If any element is out of order, has a length of 1-byte or shorter, or more than one element has the same type byte, client software **MUST** return `-32602: Invalid params` error.". 
I interpreted the "If any element ... has a length of 1-byte or shorter" as "if there is not data"

---

Hmm indeed it's confusing. The EIP mentions skipping hough, which is different from failing hard. The case of only a single byte is indeed the case where data is empty because the request type is 1 byte length.
Have you verified this against any test or client's code?

---

That section of the EIP is concerned with calculating the commitment, not the engine API. Still a good idea to check hive and other client's code though.